airbyte-commons-protocol/src/main/java/io/airbyte/commons/protocol/migrations/AirbyteMessageMigrationV1.java

```diff
@@ -4,30 +4,69 @@
 
 package io.airbyte.commons.protocol.migrations;
 
+import static io.airbyte.protocol.models.JsonSchemaReferenceTypes.REF_KEY;
+
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.node.ArrayNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import com.fasterxml.jackson.databind.node.TextNode;
 import io.airbyte.commons.json.Jsons;
 import io.airbyte.commons.version.AirbyteProtocolVersion;
 import io.airbyte.commons.version.Version;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.ConfiguredAirbyteStream;
 import io.airbyte.protocol.models.JsonSchemaReferenceTypes;
+import io.airbyte.protocol.models.v0.AirbyteRecordMessage;
 import io.airbyte.protocol.models.v1.AirbyteMessage;
+import io.airbyte.protocol.models.v1.AirbyteMessage.Type;
 import io.airbyte.protocol.models.v1.AirbyteStream;
+import io.airbyte.validation.json.JsonSchemaValidator;
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
 import java.util.function.Consumer;
 import java.util.function.Function;
 import java.util.stream.StreamSupport;
 
 public class AirbyteMessageMigrationV1 implements AirbyteMessageMigration<io.airbyte.protocol.models.v0.AirbyteMessage, AirbyteMessage> {
```

Comment:
Just seeing now that the previous PR should have added an `@Singleton` annotation here.
I'll add it to the base PR which may change some behaviors, mostly, `JsonSchemaValidator` will need a factory method.

Reply:
by factory method - just something like `public static JsonSchemaValidator build() {...}` ? or does it need to have a specific signature? and/or literally have some `class JsonSchemaValidatorFactory {...}`